packages/gatsby-cli/src/handlers/plugin-babel-utils.ts

```diff
@@ -0,0 +1,319 @@
+import * as t from "@babel/types"
+import generate from "@babel/generator"
+import template from "@babel/template"
+import { declare } from "@babel/helper-plugin-utils"
+
+const getKeyNameFromAttribute = (node: any): any =>
+  node.key.name || node.key.value
+
+const unwrapTemplateLiteral = (str: string): string =>
+  str.trim().replace(/^`/, ``).replace(/`$/, ``)
+
+const isLiteral = (node: any): boolean =>
+  t.isLiteral(node) || t.isStringLiteral(node) || t.isNumericLiteral(node)
+
+const getObjectFromNode = (nodeValue: any): any => {
+  if (!nodeValue || !nodeValue.properties) {
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define
+    return getValueFromNode(nodeValue)
+  }
+
+  const props = nodeValue.properties.reduce((acc, curr) => {
+    let value = null
+
+    if (curr.value) {
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
+      value = getValueFromNode(curr.value)
+    } else if (t.isObjectExpression(curr.value)) {
+      value = curr.value.expression.properties.reduce((acc, curr) => {
+        acc[getKeyNameFromAttribute(curr)] = getObjectFromNode(curr)
+        return acc
+      }, {})
+    } else {
+      throw new Error(`Did not recognize ${curr}`)
+    }
+
+    acc[getKeyNameFromAttribute(curr)] = value
+    return acc
+  }, {})
+
+  return props
+}
+
+const getValueFromNode = (node: any): any => {
+  if (t.isTemplateLiteral(node)) {
+    // @ts-ignore - fix me
```

Comment:
This file has a lot of `// @ts-ignore`. If you feel strongly about it I can spend time fixing it but I just wanted it to get working 😬